### PR TITLE
fix(utils): undefined when omiting prop default

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/utils/tests/prop.spec.ts [hunk 1]
```diff
@@ -437,3 +439,15 @@ describe('buildProps', () => {
     }>()
   })
 })
+
+describe('runtime', () => {
+  it('default value', () => {
+    const Foo = defineComponent({
```
Comment (HerringtonDarkholme) [inline packages/utils/tests/prop.spec.ts:445]: Can we add a test for option that has `values` but not `default`. And pass `undefeind` as runtime prop? 
It should warn the undefined value.

Comment (sxzz) [reply packages/utils/tests/prop.spec.ts:445]: I added this test. You should pass both `values` and `required`, and then it warns.

File: packages/utils/props.ts [hunk 2]
```diff
@@ -139,17 +142,19 @@ export function buildProp<
         }
       : undefined
 
-  return {
+  const prop: any = {
     type:
       typeof type === 'object' &&
```
Comment (JeremyWuuuuu) [inline packages/utils/props.ts:147]: Can we use `isObject(type)` here?

Comment (sxzz) [reply packages/utils/props.ts]: Fixed.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**